zerver/tests/test_custom_profile_data.py

```diff
@@ -454,6 +454,69 @@ def test_delete_internals(self) -> None:
         self.assertFalse(self.custom_field_exists_in_realm(field.id))
         self.assertEqual(user_profile.customprofilefieldvalue_set.count(), self.original_count - 1)
 
+    def test_delete_value_with_editable_by_user(self) -> None:
+        iago = self.example_user("iago")
+        hamlet = self.example_user("hamlet")
+        realm = iago.realm
+        self.login("hamlet")
+
+        biography_custom_field = CustomProfileField.objects.get(name="Biography", realm=realm)
+        birthday_custom_field = CustomProfileField.objects.get(name="Birthday", realm=realm)
+
+        # Set and assert our initial state.
+        data = {}
+        data["editable_by_user"] = "false"
+        result = self.api_patch(
+            iago, f"/api/v1/realm/profile_fields/{birthday_custom_field.id}", info=data
+        )
+        self.assert_json_success(result)
+
+        birthday_custom_field.refresh_from_db()
+        self.assertFalse(birthday_custom_field.editable_by_user)
+        self.assertTrue(biography_custom_field.editable_by_user)
+
+        CustomProfileFieldValue.objects.get(user_profile=iago, field=birthday_custom_field)
```

Comment:
The values are not stored in any variable and hence are not used. So, this can be deleted.

Reply:
Aren't these needed to assert that the value actually exists before we delete them? I removed one `get` here that is not deleted below.

Reply:
I agree. Updated to use `assertTrue` as suggested.